low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py

```diff
@@ -73,34 +82,71 @@ def get_health_info(self, rpath):
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
         info = {}
-        fru_found = False
+        resource_found = False
         nodes = rpath.strip().split(">")
         leaf_node, _ = self.get_node_details(nodes[-1])
+
+        # Fetch health information for all sub nodes
         if leaf_node == "compute":
-            for fru in self.server_frus:
+            info = self.get_server_health_info()
+            resource_found = True
+        elif leaf_node in self.server_resources:
+            for resource, method in self.server_resources[leaf_node].items():
                 try:
-                    info.update({fru: self.server_frus[fru]()})
+                    info.update({resource: method()})
+                    resource_found = True
                 except:
                     # TODO: Log the exception
-                    info.update({fru: None})
-            info["last_updated"] = int(time.time())
-            fru_found = True
+                    info = None
         else:
-            fru = None
             for node in nodes:
-                fru, _ = self.get_node_details(node)
-                if self.server_frus.get(fru):
-                    fru_found = True
+                resource, _ = self.get_node_details(node)
+                for res_type in self.server_resources:
+                    method = self.server_resources[res_type].get(resource)
+                    if not method:
+                        continue
                     try:
-                        info = self.server_frus[fru]()
+                        info = method()
+                        resource_found = True
                     except:
                         # TODO: Log the exception
                         info = None
+                if resource_found:
                     break
-        if not fru_found:
+
+        if not resource_found:
             raise ResourceMapError(
                 errno.EINVAL,
-                f"Health provider doesn't have support for'{rpath}'.")
+                f"Invalid rpath or health provider doesn't have support for'{rpath}'.")
+
+        return info
+
+    def get_server_health_info(self):
+        unhealthy_resource_found = False
+        server_details = Utility.get_server_details()
+        info = {}
+        info["make"] = server_details["Board Mfg"]
+        info["model"]= server_details["Product Name"]
+        for res_type in self.server_resources:
+            info.update({res_type: {}})
+            for fru, method in self.server_resources[res_type].items():
+                try:
+                    info[res_type].update({fru: method()})
+                    for data in info[res_type][fru]:
+                        if data["health"]["status"] not in ["OK", "ok"]:
+                            unhealthy_resource_found = True
```

Comment:
this derived health is one very imp item, glad you covered this

Reply:
Okay

Reply:
This is the place where cpu health response parsing is failing due to unique its schema.
@cdeshmukh @satish-seagate @jjxsg

Reply:
Ok